### PR TITLE
Feature - Standardization

### PR DESCRIPTION
**Pull request by omar-abdelgawad**
###### Ticket(s): [#XXX](https://app.jira.com/XXX)

- 🔄 **Main changes in this PR**
    - added standardization module as part of preprocessing.
    - created tests_data folder and a preprocessing folder inside it.
    -  added two sample files inside tests_data/preprocessing.
    - added pytest to requirements.
    - created an elementary test for standardization module. 

- ⚠️ **Breaking changes** 
  - Created tests_data folder inside the tests dire

## 🚀 How to test this Pull Request

1. one test is included in tests/preprocessing/test_standardize.py

## 📝 Notes for future improvements

- More tests are going to be added in the future.

### CODE REVIEW

File: decompile/preprossing/standardize.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+def standardize_asm_file(asm_file_path: str) -> str:
+    """Standardize asm file.
+
+    Args:
+        asm_file_path(str): path to asm file.
+
+    Returns:
+        str: standardized asm file as text."""
+    section_start_text = "Disassembly of section"
+    memory_address_format_example = "48 c7 45 f8 00 00 00    "
```
Comment (ahmedheakl) [inline decompile/preprossing/standardize.py:14]: Write a comment explaining why are you using this `memory_address_format_example` for.

File: requirements.txt [hunk 2]
```diff
@@ -2,3 +2,4 @@ peft
 accelerate
 transformers
 trl
+pytest
```
Comment (ahmedheakl) [inline requirements.txt:5]: `pytest` is a dev requirement not a production requirement. Please remove it.

File: decompile/preprossing/standardize.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                continue
+            if line.startswith("<"):
+                standardized_asm_buffer += line + "\n"
+            else:
+                line = line[len(memory_address_format_example) - 2 :]
+                if not line:
+                    continue
+                components = re.split(instruction_split_regex, line)
+                components = [c for c in components if c]
+                line = " ".join(components)
```
Comment (ahmedheakl) [inline decompile/preprossing/standardize.py:30]: You can remove excess spaces by the following regex
```python
line = re.sub(r"\s+", " ", line)
```

File: decompile/preprossing/standardize.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def standardize_asm_file(asm_file_path: str) -> str:
+    """Standardize asm file.
+
+    Args:
+        asm_file_path(str): path to asm file.
+
+    Returns:
+        str: standardized asm file as text."""
```
Comment (ahmedheakl) [inline decompile/preprossing/standardize.py:12]: Change to
```python
str: standardized asm file as text.
"""
```

**Review by ahmedheakl — CHANGES_REQUESTED** [review]
Really nice work :clap:. Please fix pending changes and merge into `main`. 
Also, use [black](https://black.readthedocs.io/en/stable/) for formatting, [pylint](https://pypi.org/project/pylint/) to detect liniting errors, and [mypy](https://pypi.org/project/mypy/
) for typing errors.

File: decompile/preprossing/standardize.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    section_start_text = "Disassembly of section"
+    memory_address_format_example = "48 c7 45 f8 00 00 00    "
+    instruction_split_regex = " "
+    standardized_asm_buffer = ""
+    with open(asm_file_path, "r") as f:
```
[the remark this answers is not in the document]
Comment (omar-abdelgawad) [reply decompile/preprossing/standardize.py:17]: I thought "utf-8" was the default encoding for open.
+ should I resolve a conversation after changing the files?

File: decompile/preprossing/standardize.py (hunk 1 above)
Comment (omar-abdelgawad) [reply decompile/preprossing/standardize.py:14]: Noted.

File: requirements.txt (hunk 2 above)
Comment (omar-abdelgawad) [reply requirements.txt:5]: then should I create a requirements-dev.txt? 
Also should I include mypy with pylint in it?

File: decompile/preprossing/standardize.py (hunk 3 above)
Comment (omar-abdelgawad) [reply decompile/preprossing/standardize.py:30]: Noted.

File: decompile/preprossing/standardize.py (hunk 4 above)
Comment (omar-abdelgawad) [reply decompile/preprossing/standardize.py:12]: Noted.

**Review by ahmedheakl — APPROVED** [review]
LGTM!!

**Pull request merged**